internal/report/create.go

```diff
@@ -0,0 +1,140 @@
+			Summary: v1beta1.PolicyReportSummary{
+				Pass:  0,          // count of policies with requirements met
+				Fail:  0,          // count of policies with requirements not met
+				Warn:  0,          // not used for now
+				Error: 0,          // count of policies that couldn't be evaluated
+				Skip:  skippedNum, // count of policies that were not selected for evaluation
+			},
+			Results: []*v1beta1.PolicyReportResult{},
+		}
+	}
+	return nil
+}
+
+func Print() {
+	marshaled, err := json.Marshal(clusterPolicyReport)
+	if err != nil {
+		log.Error().Err(err).Msg("marshalling error")
+	}
+	fmt.Println(string(marshaled))
+	for _, report := range policyReports {
+		marshaled, err := json.Marshal(report)
+		if err != nil {
+			log.Error().Err(err).Msg("marshalling error")
+		}
+		fmt.Println(string(marshaled))
+	}
+}
+
+func AddResult(namespace string, policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) {
+	result := newResult(policy, resource, auditResponse, responseErr)
+	switch result.Result {
+	case "fail":
+		policyReports[namespace].Summary.Fail++
+	case "error":
+		policyReports[namespace].Summary.Error++
+	case "pass":
+		policyReports[namespace].Summary.Pass++
+	}
+	policyReports[namespace].Results = append(policyReports[namespace].Results, result)
+}
+
+func newResult(policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) *v1beta1.PolicyReportResult {
+	var result v1beta1.PolicyResult
+	var description string
+	if responseErr != nil {
+		result = "error"
+		description = auditResponse.Response.Result.Message
+	} else {
+		if auditResponse.Response.Allowed {
+			result = "pass"
+		} else {
+			result = "fail"
+			description = auditResponse.Response.Result.Message
+		}
+	}
+
+	var name string
+	switch policy.GetObjectKind().GroupVersionKind() {
+	case schema.GroupVersionKind{
+		Group:   constants.KubewardenPoliciesGroup,
+		Version: constants.KubewardenPoliciesVersion,
+		Kind:    "ClusterAdmissionPolicy",
+	}:
+		name = "cap-" + policy.GetName()
+	case schema.GroupVersionKind{
+		Group:   constants.KubewardenPoliciesGroup,
+		Version: constants.KubewardenPoliciesVersion,
+		Kind:    "AdmissionPolicy",
+	}:
+		name = "ap-" + policy.GetName()
+	}
+
+	time := metav1.Now()
+	timestamp := *time.ProtoTime()
+
+	log.Debug().Any("resource", resource)
+
+	return &v1beta1.PolicyReportResult{
+		Source: PolicyReportSource,
+		Policy: name, // either cap-policy_name or ap-policy_name
+		// Timestamp shouldn't be used in go structs, and only gives seconds
+		// https://github.com/kubernetes/apimachinery/blob/v0.27.2/pkg/apis/meta/v1/time_proto.go#LL48C9-L48C9
+		Timestamp:   timestamp,               // time the result was found
+		Result:      result,                  // pass, fail, error
+		Subjects:    []*v1.ObjectReference{}, // TODO reference to object evaluated
```

Comment:
This is missing.

Reply:
I've pushed a commit  proposing a fix for that. ;)